src/realm/bplustree.hpp

```diff
@@ -458,6 +458,20 @@ class BPlusTree : public BPlusTreeBase {
             set(ndx1, tmp2.get());
             set(ndx2, tmp1.get());
         }
+        else if constexpr (std::is_same_v<T, Mixed>) {
+            std::string buf1;
+            std::string buf2;
+            Mixed tmp1 = get(ndx1);
+            Mixed tmp2 = get(ndx2);
+            if (tmp1.is_type(type_String, type_Binary)) {
+                tmp1.use_buffer(buf1);
+            }
+            if (tmp2.is_type(type_String)) {
```

Comment:
I think this is missing a check for  type_BInary as well. Could verify by adding a test for swap between a string/binary.

Reply:
An oversight